guardian/support-service-lambdas
# Add basic national delivery product to new-product-api

This is a basic PR to add the National delivery product.

- refactoring and simplification on Catalog.scala to separate out each of the separate products, as it was getting out of hand in that huge function.
- general cleanup based on warnings etc from intellij
- added National delivery as a clone of the other Paper

This is behind a switch - you need to pass in ?nationalDelivery=true in the URL to preview the new behaviour.

Tested in CODE
- the national delivery plans only appear in the catalog with the above switch parameter.  See [1] below for data
- it successfully created a valid looking subscription on an existing account https://apisandbox.zuora.com/apps/CustomerAccount.do?method=view&id=8ad08ae28aa80df2018aa8d1f476107d

TODO in future PRs

- add the email ID and correct fields for the thank you email
- add the delivery provider field - this will require a new way to allow CSRs to choose it.
- testing with salesforce

[1] Data as of 18th September
```
        {
            "label": "National Delivery",
            "plans": [
                {
                    "id": "national_delivery_sixday",
                    "label": "Sixday",
                    "startDateRules": {
                        "daysOfWeek": [
                            "Monday",
                            "Tuesday",
                            "Wednesday",
                            "Thursday",
                            "Friday",
                            "Saturday"
                        ],
                        "selectableWindow": {
                            "startDate": "2023-09-21",
                            "sizeInDays": 28
                        }
                    },
                    "paymentPlans": [
                        {
                            "currencyCode": "GBP",
                            "description": "GBP 64.99 every month"
                        }
                    ],
                    "paymentPlan": "GBP 64.99 every month"
                },
                {
                    "id": "national_delivery_everyday",
                    "label": "Everyday",
                    "startDateRules": {
                        "daysOfWeek": [
                            "Monday",
                            "Tuesday",
                            "Wednesday",
                            "Thursday",
                            "Friday",
                            "Saturday",
                            "Sunday"
                        ],
                        "selectableWindow": {
                            "startDate": "2023-09-21",
                            "sizeInDays": 28
                        }
                    },
                    "paymentPlans": [
                        {
                            "currencyCode": "GBP",
                            "description": "GBP 73.99 every month"
                        }
                    ],
                    "paymentPlan": "GBP 73.99 every month"
                },
                {
                    "id": "national_delivery_weekend",
                    "label": "Weekend",
                    "startDateRules": {
                        "daysOfWeek": [
                            "Saturday",
                            "Sunday"
                        ],
                        "selectableWindow": {
                            "startDate": "2023-09-23",
                            "sizeInDays": 28
                        }
                    },
                    "paymentPlans": [
                        {
                            "currencyCode": "GBP",
                            "description": "GBP 29.99 every month"
                        }
                    ],
                    "paymentPlan": "GBP 29.99 every month"
                }
            ],
            "enabledForDeliveryCountries": [
                "United Kingdom"
            ]
        }
```
